tools/src/test/java/org/wso2/carbon/tools/dropins/DropinsDeployerToolTest.java

```diff
@@ -0,0 +1,184 @@
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * This class defines the unit test cases for DropinsDeployerToolUtils.java.
+ *
+ * @since 5.1.0
+ */
+public class DropinsDeployerToolTest {
+    private static final String testArtifactOne = "org.wso2.carbon.sample.datasource.mgt.jar";
+    private static final String testArtifactTwo = "org.wso2.carbon.sample.dbs.deployer.jar";
+    private static final String testArtifactThree = "org.wso2.carbon.sample.deployer.mgt.jar";
+    private static final String testArtifactFour = "tool-test-artifact.jar";
+    private static final Path carbonHome = Paths.get("target", "carbon-home");
+    private static final List<String> profileNames = new ArrayList<>();
+
+    @BeforeClass
+    public static void initTestClass() throws IOException {
+        createDirectories(carbonHome);
+    }
+
+    @Test(description = "Attempts to get Carbon profiles when no profiles directory exists", expectedExceptions = {
+            CarbonToolException.class }, priority = 0)
+    public void testGettingProfilesWhenNoProfilesDirectory() throws CarbonToolException {
+        DropinsDeployerToolUtils.executeTool(carbonHome.toString());
+    }
+
+    @Test(description = "Attempts to retrieve a string notifying the Carbon profiles available when no profiles "
+            + "are available", priority = 1)
+    public void testRetrievingProfileStringWhenNoProfileExists() throws IOException {
+        prepareCarbonHomeForDropinsTests();
+        StringBuilder actual = DropinsDeployerToolUtils.getProfileString(carbonHome.toString());
+        Assert.assertTrue("WSO2 CARBON PROFILES\nNo profiles available".equals(actual.toString()));
+    }
+
+    @Test(description = "Attempts to retrieve an non-existent Carbon Profile", priority = 2)
+    public void testChoosingNonExistingCarbonProfile() throws IOException {
+        Optional<String> profileName = DropinsDeployerToolUtils.getUserChoice(carbonHome.toString(), 7);
+        Assert.assertTrue(!profileName.isPresent());
+    }
+
+    @Test(description = "Attempts to execute dropins capability with all available Carbon Profiles", priority = 3)
+    public void testExecutingDropinsCapabilityWithAllProfiles() throws CarbonToolException, IOException {
+        createProfiles();
+        ByteArrayInputStream stream = new ByteArrayInputStream(("" + (profileNames.size() + 1)).getBytes());
+        System.setIn(stream);
+
+        List<BundleInfo> expected = getExpectedBundleInfo();
+        List<BundleInfo> actual;
+
+        System.setProperty("wso2.carbon.tool", "dropins-deployer");
+        String[] args = { carbonHome.toString() };
+        CarbonToolManager.main(args);
+
+        for (String profileName : profileNames) {
+            Path bundlesInfo = Paths.get(carbonHome.toString(), "osgi", "profiles", profileName, "configuration",
```

Comment:
"osgi" folder name is used in many places. Won't it be better to have it as a Constant? Same goes for the other folder names.